routes/favicon.coffee

```diff
@@ -0,0 +1,27 @@
+request = require 'request'
+fs = require 'fs'
+tmpdir = require 'os'
+	.tmpdir()
+path = require 'path'
+CACHEFILE = 'favicon.ico'
+
+module.exports = (req, res, next) ->
+	res.sendFile CACHEFILE,
+		root: tmpdir
+		(error) ->
+			if error
```

Comment:
I normally check for errors with the `?` existential operator too. `return unless error?`